engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -700,7 +696,7 @@ class Compiler(
     * @return A Tree representation of `source`
     */
   def parseInline(source: CharSequence): Tree =
-    ensoCompiler.parse(source)
+    Parser.parse(source)
```

Comment:
Having `EnsoParser` a few lines above and here `Parser` it really begs a question: "What's the difference?"/"Which one should I use?" Wouldn't it better for `EnsoParser.parse` to simply forward to `Parser.parse` to hide that?

Reply:
I prefer not to expose two different levels of abstraction from `EnsoParser`, so that `EnsoParser` functions output `IR` but not `Tree`. Backend parser-users work with `IR` exclusively, except the single caller of this `parseInline` function. The caller uses the `Tree` API to validate whether an input is *inline*. If we were to rewrite those few lines of code to use the `IR` API, we would be able to treat the entire `Tree` API as an internal implementation detail of the parser.